### PR TITLE
Refactor order processing logic

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/EventListener/OrderRecalculationListener.php [hunk 1]
```diff
@@ -24,14 +23,14 @@
...
     private $orderRecalculator;
 
     /**
-     * @param OrderRecalculatorInterface $orderRecalculator
+     * @param OrderProcessorInterface $orderRecalculator
      */
-    public function __construct(OrderRecalculatorInterface $orderRecalculator)
+    public function __construct(OrderProcessorInterface $orderRecalculator)
     {
         $this->orderRecalculator = $orderRecalculator;
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderRecalculationListener.php]: Invalid variable.

File: src/Sylius/Bundle/CoreBundle/EventListener/OrderShippingListener.php [hunk 2]
```diff
@@ -21,24 +21,24 @@
 /**
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  */
-class OrderShippingListener
+final class OrderShippingListener
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderShippingListener.php]: This entire listener should probably be removed now.

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderShippingListener.php]: You're right, I'm fixing the specs now and removing this class was the best way to fix the related spec :D

File: src/Sylius/Component/Core/spec/OrderProcessing/ShippingChargesProcessorSpec.php [hunk 3]
```diff
@@ -57,12 +58,12 @@ function it_doesnt_apply_any_shipping_charge_if_order_has_no_shipments(OrderInte
...
         $order->addAdjustment(Argument::any())->shouldNotBeCalled();
 
-        $this->applyShippingCharges($order);
+        $this->process($order);
     }
 
     function it_applies_calculated_shipping_charge_for_each_shipment_associated_with_the_order(
         FactoryInterface $adjustmentFactory,
-        $calculator,
+        DelegatingCalculatorInterface$calculator,
```
Comment (Arminek) [inline src/Sylius/Component/Core/spec/OrderProcessing/ShippingChargesProcessorSpec.php]: Missing space

File: src/Sylius/Component/Core/OrderProcessing/CompositeOrderProcessor.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\OrderProcessing;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Zend\Stdlib\PriorityQueue;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
```
Comment (Zales0123) [inline src/Sylius/Component/Core/OrderProcessing/CompositeOrderProcessor.php]: I don't think so 😄

File: src/Sylius/Component/Core/spec/OrderProcessing/ShippingChargesProcessorSpec.php (hunk 3 above)
Comment (pamil) [inline src/Sylius/Component/Core/spec/OrderProcessing/ShippingChargesProcessorSpec.php]: It's not a bug, it's a feature! :tada:

**Comment by pjedrzejewski** [comment]
Great work Kamil, thanks! ;)

**Pull request merged**